src/containers/Masthead/components/MastheadSearch.tsx

```diff
@@ -39,48 +37,40 @@ interface Props {
   subject?: GQLMastheadSearch_SubjectFragment;
 }
 
-const StyledButton = styled(ButtonV2)`
-  padding: ${spacing.small} ${spacing.normal};
-  gap: ${spacing.medium};
-  svg {
-    width: 24px;
-    height: 24px;
-  }
-
-  ${mq.range({ until: breakpoints.mobileWide })} {
-    border-radius: 100%;
-    background: transparent;
-    border-color: transparent;
-    padding: ${spacing.xsmall};
-    span {
-      display: none;
-    }
-  }
-`;
-
-const StyledCloseButton = styled(IconButtonV2)`
-  margin-top: ${spacing.xsmall};
-`;
-
-const SearchWrapper = styled.div`
-  width: 100%;
-  padding: ${spacing.normal} ${spacing.normal};
-  display: flex;
-  align-items: flex-start;
-  gap: ${spacing.xsmall};
-  ${mq.range({ from: breakpoints.desktop })} {
-    width: 60%;
-  }
-`;
-
-const StyledDrawer = styled(Drawer)`
-  background-color: ${colors.brand.greyLightest};
-  display: flex;
-  justify-content: center;
-`;
+const StyledButton = styled(Button, {
+  base: {
+    mobileWideDown: {
+      "& span": {
+        display: "none",
+      },
+    },
+  },
+});
+
+const SearchWrapper = styled("div", {
+  base: {
+    width: "100%",
+    paddingBlock: "medium",
+    paddingInline: "medium",
+    display: "flex",
+    alignItems: "flex-start",
+    gap: "3xsmall",
+    desktop: {
+      width: "60%",
+    },
+  },
+});
+
+const StyledDialogContent = styled(DialogContent, {
+  base: {
+    display: "flex",
+    justifyContent: "center",
+    height: "unset",
+  },
+});
 
 const MastheadSearch = ({ subject }: Props) => {
-  const [isOpen, setIsOpen] = useState(false);
+  const [dialogState, setDialogState] = useState({ open: false });
```

Comment:
Om root trenger et objekt for `onOpenChange` kan vi typesette dette på en måte? Så vi vet hva som kan være i `dialogState`?

Reply:
Ark eksporterer typer for det, men lurer på om vi skal vente litt med det. Litt usikker på hva konvensjonen for state skal være med de nye komponentene inntil videre :')